charms/knative-serving/src/image_management.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+from typing import Dict
+
+import yaml
+
+
+def parse_image_config(image_config: str) -> Dict[str, str]:
+    """Parses config data for a dict of images, returning the parsed value as a dict.
+
+    This helper does not catch YAML parsing errors, leaving that up to the calling function.
+    Generally, this should be called inside a `try/except YAMLError` block.
+    """
+    images_from_config = yaml.safe_load(image_config)
+
+    if not images_from_config:
+        images_from_config = {}
+
+    images_from_config = remove_empty_images(images=images_from_config)
+
+    return images_from_config
+
+
+def remove_empty_images(images: Dict[str, str]):
+    """Removes any image with a value of an empty string."""
```

Comment:
Please change the docstring to reflect the return of this method, if possible also add its return type

Reply:
updated!